src/SAML2/Attribute/Attribute.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Surfnet\SamlBundle\SAML2\Attribute;
+
+use Serializable;
+
+class Attribute implements Serializable
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var array
+     */
+    private $values;
+
+    /**
+     * @var string
+     */
+    private $urnMace;
+
+    /**
+     * @var string
+     */
+    private $urnOid;
+
+    /**
+     * @param AttributeDefinition $definition
+     * @param array $values
+     */
+    public function __construct(AttributeDefinition $definition, array $values)
+    {
+        $this->name = $definition->getName();
+        $this->urnMace = $definition->getUrnMace();
+        $this->urnOid = $definition->getUrnOid();
+        $this->values = $values;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUrnMace()
+    {
+        return $this->urnMace;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUrnOid()
+    {
+        return $this->urnOid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValues()
+    {
+        return $this->values;
+    }
+
+    /**
+     * @param Attribute $attribute
+     * @return bool
+     */
+    public function equals(Attribute $attribute)
+    {
+        return $attribute == $this;
```

Comment:
hrms, even though it is correct, not sure I like this. Especially with my remark about using the definition object. For these kind of checks I favour explicitness. Personally I'd prefer something like

`return $this->definition->equals($other->definition) && $this->value === $other->value`

as it delegates definition equality check to the definition, and forces values to be strictly equal...